integrations/anthropic/src/haystack_integrations/components/generators/anthropic/adapter.py

```diff
@@ -0,0 +1,98 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, fields
+from typing import Any, ClassVar, Dict, Type
+
+from haystack.core.errors import DeserializationError
+from haystack.lazy_imports import LazyImport
+from haystack.utils.auth import Secret
+
+with LazyImport('Run pip install -U google-cloud-aiplatform "anthropic[vertex]".'):
+    from anthropic import AnthropicVertex
+
+
+with LazyImport("Run pip install anthropic."):
+    from anthropic import Anthropic
+
+
+@dataclass(frozen=True)
+class BaseAdapter(ABC):
+    """
+    Base class for model APIs supported by AnthropicGenerator.
+    """
+
+    TYPE: ClassVar[str]
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Converts the object to a dictionary representation for serialization.
+        """
+        _fields = {}
+        for _field in fields(self):
+            if _field.type is Secret:
+                _fields[_field.name] = getattr(self, _field.name).to_dict()
+            else:
+                _fields[_field.name] = getattr(self, _field.name)
+
+        return {"type": self.TYPE, "init_parameters": _fields}
+
+    @staticmethod
+    def from_dict(data: Dict[str, Any]) -> "BaseAdapter":
+        """
+        Converts a dictionary representation to an adapter object.
+        """
```

Comment:
This method is needed only if we want serialize the adapter. Not sure if thats needed.